### ISSUE TITLE
Release 11.1.0

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#4333]
Going to release tomorrow.

- [x] stylelint release
- [x] stylelint-demo update
- [x] stylelint.io update
- [x] tweet

**Comment by hudochenkov** [comment]
No release for now. Out of blue [17 failed tests](https://github.com/stylelint/stylelint/runs/253528783) on Windows.

I the common theme:

```
    expect(received).toBe(expected) // Object.is equality

    Expected: "D:\\a\\stylelint\\stylelint\\lib\\__tests__\\fixtures\\empty-block-with-disables.css"
    Received: "d:\\a\\stylelint\\stylelint\\lib\\__tests__\\fixtures\\empty-block-with-disables.css"
```

Looks like path that known to stylelint and path which we get from Node.js `path` in tests, are different.

Maybe one of our dependencies had release in the past 5 hours and it broke something. 5 hours ago we had green CI.

**Comment by hudochenkov** [comment]
I wish we had `package-lock.json` to confirm this :(

**Comment by evilebottnawi** [comment]
@hudochenkov maybe it is time to lock file? I have been proposing to do this for a long time :smile:

**Comment by vankop** [comment]
It is dep of one of out dep as I see =) because I can not find any released one

**Comment by vankop** [comment]
As I understand current `11.0.0` also affected

**Comment by hudochenkov** [comment]
Does anyone has local stylelint repository with `node_modules` installed at least 7 hours ago? Could you remove `.npmrc` and run `npm i --package-lock-only` to create package-lock and then share it with me?

**Comment by vankop** [comment]
https://gist.github.com/vankop/07c4967f0d5abe3a22c764ab93ec5d81

**Comment by vankop** [comment]
oh sorry it is not master =(

**Comment by vankop** [comment]
latest changes for me
<img width="914" alt="Снимок экрана 2019-10-09 в 16 06 36" src="https://user-images.githubusercontent.com/10380560/66484040-e6dd9080-eaae-11e9-8792-e871ec4e2b03.png">

**Comment by hudochenkov** [comment]
It might be Github Actions fault. Failing are inconsistent. Sometimes they happen, sometimes — not. No restart for master latest commit fix the build. But restarts in #4334 fixed the build once.

What I noticed between successful runs and failed, that Github Actions setup environment with different paths!

On success it uses `D:\`, on fails it uses `d:\`.

<details>
<summary>Fail</summary>

```
Run actions/checkout@v1

remote: Counting objects:   9% (16/175)        
Run actions/checkout@v1
Added matchers: 'checkout-git'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]d:\a\_temp\git_794fe967-2362-44f8-b1e3-29892fc485ec.json
Syncing repository: stylelint/stylelint
git version
git version 2.23.0.windows.1
git lfs version
git-lfs/2.8.0 (GitHub; windows amd64; go 1.12.2; git 30af66bb)
git init "d:\a\stylelint\stylelint"
Initialized empty Git repository in d:/a/stylelint/stylelint/.git/

-----------------

Use Node.js 10

##[add-matcher]d:\a\_actions\actions\setup-node\v1\.github\eslint-compact.json
Run actions/setup-node@v1
##[add-path]C:\hostedtoolcache\windows\node\10.16.3\x64
Added matchers: 'tsc'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]d:\a\_actions\actions\setup-node\v1\.github\tsc.json
Added matchers: 'eslint-stylish'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]d:\a\_actions\actions\setup-node\v1\.github\eslint-stylish.json
Added matchers: 'eslint-compact'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]d:\a\_actions\actions\setup-node\v1\.github\eslint-compact.json

-------------

Install latest npm

added 430 packages from 839 contributors in 41.564s
Run npm install --global npm@latest
C:\npm\prefix\npm -> C:\npm\prefix\node_modules\npm\bin\npm-cli.js
C:\npm\prefix\npx -> C:\npm\prefix\node_modules\npm\bin\npx-cli.js
```
</details>


<details>
<summary>Success</summary>

```
Run actions/checkout@v1

##[remove-matcher owner=checkout-git]
Run actions/checkout@v1
Added matchers: 'checkout-git'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]D:\a\_temp\git_d099a675-108d-4641-a091-932d1275b73d.json
Syncing repository: stylelint/stylelint
git version
git version 2.23.0.windows.1
git lfs version
git-lfs/2.8.0 (GitHub; windows amd64; go 1.12.2; git 30af66bb)
git init "D:\a\stylelint\stylelint"
Initialized empty Git repository in D:/a/stylelint/stylelint/.git/

----------------------------

Use Node.js 10

##[add-matcher]D:\a\_actions\actions\setup-node\v1\.github\eslint-compact.json
Run actions/setup-node@v1
##[add-path]C:\hostedtoolcache\windows\node\10.16.3\x64
Added matchers: 'tsc'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]D:\a\_actions\actions\setup-node\v1\.github\tsc.json
Added matchers: 'eslint-stylish'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]D:\a\_actions\actions\setup-node\v1\.github\eslint-stylish.json
Added matchers: 'eslint-compact'. Problem matchers scan action output for known warning or error strings and report these inline.
##[add-matcher]D:\a\_actions\actions\setup-node\v1\.github\eslint-compact.json

---------------------

Install latest npm

added 430 packages from 839 contributors in 25.741s
Run npm install --global npm@latest
C:\npm\prefix\npx -> C:\npm\prefix\node_modules\npm\bin\npx-cli.js
C:\npm\prefix\npm -> C:\npm\prefix\node_modules\npm\bin\npm-cli.js
```
</details>

**Comment by vankop** [comment]
I can not find this code in JS https://github.com/nodejs/node/blob/master/lib/fs.js#L1418 , but it possible that C/C++ already returns path with different cases

**Comment by vankop** [comment]
> Looks like path that known to stylelint and path which we get from Node.js `path` in tests, are different.

I have checked tests, this is very strange behavior. I think exact behavior is that:
- our code **always** path same absolute path
- node interprets path as correct, because in Windows path are case insensitive (normally).
- in expect case we check path as case sensitive (somehow we get correct path)

**Comment by hudochenkov** [comment]
I did a lot of investigation and the problem lies in two places.

First, Github Actions randomly use `d:` or `D:` as drive letter, at the same time they always have Node.js installed on `C:` drive.

Second, Node.js butchering file path:

```js
console.log(process.cwd());
// 'd:\\a\\stylelint\\stylelint'

console.log(__dirname);
// 'D:\\a\\stylelint\\stylelint\\lib\\__tests__'

console.log({ path.join(__dirname, 'fixtures'));
// 'D:\\a\\stylelint\\stylelint\\lib\\__tests__\\fixtures'
```

What solution I see is to contact Github support and to ask make drive letter always uppercase. It's probably a bug, that it's sometimes lowercased. I'm going to contact them tomorrow.

**Comment by vankop** [comment]
As I mentioned above it does not make sense for Windows, because all paths in Windows are case insensitive (so it is our problem, if we check it as case sensitive or one of our dependency returns different path). Anyway it will be great, if github support will clarify this.

**Comment by vankop** [comment]
> Second, Node.js butchering file path:
> 
> ```js
> console.log(process.cwd());
> // 'd:\\a\\stylelint\\stylelint'
> 
> console.log(__dirname);
> // 'D:\\a\\stylelint\\stylelint\\lib\\__tests__'
> 
> console.log({ path.join(__dirname, 'fixtures'));
> // 'D:\\a\\stylelint\\stylelint\\lib\\__tests__\\fixtures'
> ```

process.cwd making system call each time, so there is no Node.js stuff behind

**Comment by hudochenkov** [comment]
I contacted Github support. Hoping for the fast resolution.

**Comment by hudochenkov** [comment]
Because tests are actually ok, and things are breaking because of inconsistent Github Actions environment. I'm going to release stylelint.

**Comment by hudochenkov** [comment]
New version released, but I'm not going to tweet yet. Looks like I messed up a little bit (https://github.com/stylelint/stylelint-demo/pull/177#issuecomment-540594505).
